modules/auxiliary/gather/query_ldap.rb

```diff
@@ -0,0 +1,279 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::LDAP
+  require 'json'
+  require 'yaml'
+
+  def initialize(info = {})
+    begin
+      @default_settings_file_path = ::File.join(::Msf::Config.data_directory, 'auxiliary', 'gather', 'query_ldap', 'default.yaml')
+      @default_settings = YAML.safe_load_file(@default_settings_file_path)
+    rescue StandardError => e
+      print_error("Couldn't parse #{@default_settings_file_path}, error was: #{e}")
+      return
+    end
+
+    unless @default_settings['queries']&.class == Array && !@default_settings['queries'].empty?
+      print_error("No queries supplied in #{@default_settings_file_path}!")
+      return
+    end
+
+    actions = []
+    for entry in @default_settings['queries']
+      if entry['action'].nil? || entry['description'].nil?
+        print_warning("Invalid entry detected, check the format of the file at #{@default_settings_file_path}!")
+        next
+      end
+      actions << [entry['action'], { 'Description' => entry['description'] }]
+    end
+    actions << ['RUN_QUERY_FILE', { 'Description' => 'Execute a custom set of LDAP queries from the JSON or YAML file specified by QUERY_FILE.' }]
+    actions.sort!
+
+    super(
+      update_info(
+        info,
+        'Name' => 'LDAP Query and Enumeration Module',
+        'Description' => %q{
+          This module allows users to query an LDAP server using either a custom LDAP query, or
+          a set of LDAP queries under a specific category. Users can also specify a JSON or YAML file containing
+          custom queries to be executed using the RUN_QUERY_FILE action. If this action is specified,
+          then QUERY_FILE_PATH must be a path to the location of this JSON/YAML file on disk.
+
+          Alternatively one can run one of several predefined queries by setting ACTION to the
+          appropriate value.
+
+          All results will be returned to the user in table format, with || as the delimiter
+          separating multiple items within one column.
+        },
+        'Author' => [
+          'Grant Willcox', # Module
+        ],
+        'References' => [
+        ],
+        'DisclosureDate' => '2022-05-19',
+        'License' => MSF_LICENSE,
+        'Actions' => actions,
+        'DefaultAction' => 'ENUM_ALL_OBJECTCLASS',
+        'DefaultOptions' => {
+          'SSL' => false
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS],
+          'Reliability' => []
+        }
+      )
+    )
+
+    register_options([
+      Opt::RPORT(389), # Set to 636 for SSL/TLS
+      OptEnum.new('OUTPUT_FORMAT', [true, 'The output format to use', 'table', ['csv', 'table', 'json']]),
+      OptString.new('BASE_DN', [false, 'LDAP base DN if you already have it']),
```

Comment:
It'd be great to add support for providing an inline filter + list of attributes

Reply:
Added support in with c7c7d16ee8780e9fa51be861c36e8b9a3d8c42d8, thanks for the recommendation!